lib/dradis/plugins/templates.rb

```diff
@@ -40,16 +83,81 @@ def plugin_templates(args={})
+
+          Mapping.find_or_create_by!(
+            component: @integration_name,
+            source: mapping_source,
+            destination: destination
+          )
+        end
+
+        def create_mapping_fields(mapping, template_file)
+          template_fields = parse_template_fields(template_file)
+
+          # create a mapping_field for each field in the .template file
+          template_fields.each do |field_title, field_content|
+            # set source_field by taking the first match to the existing %% syntax
+            source_field = field_content.match(LEGACY_FIELDS_REGEX)
+            source_field = source_field ? source_field[1] : 'custom text'
+
+            updated_content = update_syntax(field_content)
+
+            mapping.mapping_fields.find_or_create_by!(
+              source_field: source_field,
+              destination_field: field_title,
+              content: updated_content
+            )
+          end
+        end
+
+        def migrate(template_file, source)
+          rtp_ids = defined?(Dradis::Pro) ? ReportTemplateProperties.ids : [nil]
+          rtp_ids.each do |rtp_id|
+            @rtp_id = rtp_id
+
+            ActiveRecord::Base.transaction do
+              mapping = create_mapping(source)
+              create_mapping_fields(mapping, template_file)
+            end
+          end
+          File.rename template_file, "#{template_file}.legacy"
+        end
+
+        # previously our integrations with multiple uploaders (Burp, Qualys) had inconsistent
+        # template names (some included the uploader, some didn't ex. burp issue vs html_evidence)
+        # they have been renamed to follow a consistent 'uploader_entity' structure, but
+        # in order to migrate the old templates to the db with the new names as the source
+        # we need to reference an object in the integration that maps the new name to the old one
+        def migrate_multiple_uploaders(integration)
+          return unless LEGACY_MAPPING_REFERENCE[integration]
+
+          LEGACY_MAPPING_REFERENCE[integration].each do |source_field, legacy_template_name|
+            template_file = Dir["#{@integration_templates_dir}/#{legacy_template_name}.template*"]
+            if template_file.any? { |file| File.exist?(file) }
+              migrate(template_file[0], source_field)
+            end
+          end
+        end
+
+        def parse_template_fields(template_file)
+          template_content = File.read(template_file)
+          FieldParser.source_to_fields(template_content)
+        end
+
+        def update_syntax(field_content)
+          # turn the %% syntax into the new
+          # '{{ <integration>[was-issue.title] }}' format
+          field_content.gsub(LEGACY_FIELDS_REGEX) do |content|
+            "{{ #{@integration_name}[#{content[1..-2]}] }}"
+          end
+        end
+
+        # Skip copying the file if it's a .template or a .fields file
+        # since we are no longer using files for these
```

Comment:
The reason why this method existed was to avoid overwriting a possibly customised .template file in the app, with a fresh original from the integration. We copied everything _except_ the template.

That world doesn't exist any more. Now we only copy .sample files, anything else we can ignore. Is it time to revisit the need for this method?

Reply:
Removed and updated the `plugin_templates` method to only pull the `.sample` files